pkg/runtime/wfengine/backends/actors/workflow_actor.go

```diff
@@ -482,11 +493,14 @@ func (wf *workflowActor) runWorkflow(ctx context.Context, actorID string, remind
 		}
 		return newRecoverableError(fmt.Errorf("failed to schedule a workflow execution: %w", err))
 	}
-	// Record metrics for workflow execution
+
+	wf.recordWorkflowSchedulingLatency(ctx, isWorkflowExecutionStartedEvent, state, scheduledStartTimestamp, workflowName, executionStatus)
```

Comment:
`executionStatus` should not be used here. It is being set only when this method is returning. At this point, it would be `diag.StatusFailed` only.